### ISSUE TITLE
Feature request: arm64 win support

### ISSUE THREAD

**Issue by davidmatson** [facebookresearch/faiss#2975]
We're currently using faiss on arm64 on Windows, but we can't use the main branch as-is for that.

**Comment by mdouze** [comment]
Have you tried to compile it by hand?

**Comment by davidmatson** [comment]
Yes, I think so. We currently get it compiling by patching in edits from this branch:
https://github.com/es0m/faiss/tree/user/es0m/support-win-arm-builds
(if that's what you mean by "by hand"; I may have misunderstood)

**Comment by mdouze** [comment]
Yes sure. So IIUC the only change in the C++ part is in platform_macros.h
I think we are not going to include it the CircleCI contiuous intergation but we can accept this patch so that people can compile for arm64 windows themselves.

**Comment by davidmatson** [comment]
@mdouze - that would be great, thanks!

Yeah, I think the CircleCI part is a separate question. We consume via vcpkg, so if we could get whatever of the C++ & CMake parts needed so that it will "just work" for vcpkg consumers on Windows, that would be great.

**Comment by mdouze** [comment]
@algoriddle what do you think about the cmake patch ?

**Comment by shijunz** [comment]
> @mdouze - that would be great, thanks!
> 
> Yeah, I think the CircleCI part is a separate question. We consume via vcpkg, so if we could get whatever of the C++ & CMake parts needed so that it will "just work" for vcpkg consumers on Windows, that would be great.

yes, I try to build, it fail, anyone will help me?

**Comment by khmyznikov** [comment]
We had the [support already](https://github.com/faiss-wheels/faiss-wheels/issues/127#issuecomment-4644423885) in cpu specific package: https://github.com/faiss-wheels/faiss-wheels/pull/130 and the https://pypi.org/project/faiss-cpu/1.13.2/#files version even have the wheels published. Could you please elaborate why the support was not transferred?

@limqiying @alibeklfc
